testing/runner.ts

```diff
@@ -0,0 +1,114 @@
+import chalk from "chalk";
+import { Suite } from "df/testing";
+import { promisify } from "util";
+
+export interface IRunResult {
+  path: string[];
+  err: any;
+  outcome: "passed" | "timeout" | "failed";
+}
+
+export interface IRunContext {
+  path: string[];
+  results: IRunResult[];
+}
+
+export class Runner {
+  public static setNoExit(noExit: boolean) {
+    Runner.noExit = noExit;
+  }
+  public static queueRunAndExit() {
+    if (!Runner.resultPromise) {
+      Runner.resultPromise = Runner.run();
+    }
+  }
+
+  // tslint:disable: no-console
+  public static async run() {
+    // We tell the runner to start running at the end of current block of
+    // synchronously executed code. This will typically be after all the
+    // suite definitions are evaluated.
+    await promisify(process.nextTick)();
```

Comment:
i have no idea why exactly 1 tick is correct

Reply:
This is equivelant to setTimeout(..., 0).
It basically means - finish processing whats currently on the node event queue and then run the tests.

Reply:
sorry, yes, i know what it means, but why are we doing it on the next tick? especially since the comment says "typically", it implies it might be the case but not always (?)